packages/KPop/KPop.vue

```diff
@@ -281,7 +281,10 @@ export default {
           this.createInstance()
         }
       } else if (this.$refs.popper && this.$refs.popper.contains(e.target) && this.onPopoverClick) {
-        this.onPopoverClick()
+        const shouldShow = !this.onPopoverClick()
+        if (shouldShow !== undefined) {
+          shouldShow ? this.showPopper() : this.hidePopper()
```

Comment:
when would we want to keep it shown? also I guess we are turning this prop into one that will always control hiding and showing, which is a little less versatile I think, like there are cases where you might want to pass a callback function on click but not want to do anything with showing or hiding

Reply:
Like its useful for a dropdown implementation but we are assuming the user will always have a dropdown implementation for using this.. what if they just wanted to execute a custom callback whenever the popover is already opened? And their implementation isn't a dropdown? maybe its a checkbox or multi select or something.

Reply:
> we are turning this prop into one that will always control hiding and showing

Updated it slightly didn't realize I was accidentally checking the falsity of the callback's return. It will not always control hiding/showing. It works the exact same as your current PR except that if the callback provided returns a boolean then it will decide to hide or show. If the callback returns `void`, then it will not do anything. It allows the implementation to stay minimal without needing to know about popper or refs. I can help update some documentation if you agree this is preferable.

>  what if they just wanted to execute a custom callback whenever the popover is already opened? And their implementation isn't a dropdown? maybe its a checkbox or multi select or something.

The callback still always fires, just its return type is now meaningful.

Reply:
yeah i think having a few examples in the docs will help, in this case there's two return types that do different things. returning a boolean triggers a show or hide and returning a void just calls the callback